### DISCUSSION TITLE
Clicking a Link requests server component infinitely and does not navigate to page before manual hard refresh in production

### DISCUSSION THREAD

**Discussion by dn1t** [vercel/next.js#52554] [Help]
### Summary

When I click a Link(next/link), Next.js sends requests to `/(page)?_rsc=(rsc)` infinitely and does not navigate to the linked page unless I refresh my page manually. It does not happen in development, but just in production mode. I think my code may be wrong because I don't find similar cases when I search it and it only happens to me. I wonder if there are any common mistakes that many people make in this regard

I tried on both Next.js v13.4.9 and v13.4.10-canary.3 on Safari v17.0(19616.1.20.11.3), Firefox v116.0b2, Chrome v114.0.5735.147, macOS 14.0 Beta 3(23A5286g).

https://github.com/vercel/next.js/assets/45326612/3bfb1e07-e381-422a-a41f-f1b09aaaf0ac

### Additional information

```js
Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 23.0.0: Fri Jun 30 17:47:56 PDT 2023; root:xnu-10002.0.168.505.3~1/RELEASE_X86_64
    Binaries:
      Node: 20.4.0
      npm: 9.7.2
      Yarn: 1.22.19
      pnpm: 8.6.7
    Relevant Packages:
      next: 13.4.10-canary.3
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.2.0-beta
    Next.js Config:
      output: N/A
```

**Comment by icyJoseph** [comment]
Does this happen locally using production build?

**Reply by dn1t** [reply]
Yeah it happens both local and vercel

**Reply by icyJoseph** [reply]
mmm is this using interceptor routes? or perhaps some data fetching is failing? Are there any server async components that might be accidentally imported into a `use client` tree?

**Reply by dn1t** [reply]
I'm not using any interceptor routes and some server components are imported(maybe as client components) from client components but they're not async.

https://github.com/EntryHack/daisy

This is the repository, and navigation between all pages cause the error except when navigating between '/' and '/sticker/[id]'.

**Reply by icyJoseph** [reply]
That's a private repository :/ 

Not really sure what else could it be though, a bit clueless at this point. Are you using export mode or something like that? How is the app running (after build)?

**Comment by cherealnice** [comment]
I'm having a similar issue. Are you by chance using `generateMetadata` on your pages?
